pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/range/RangeIndexType.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.segment.index.range;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.segment.spi.ColumnMetadata;
+import org.apache.pinot.segment.spi.V1Constants;
+import org.apache.pinot.segment.spi.creator.IndexCreationContext;
+import org.apache.pinot.segment.spi.index.FieldIndexConfigs;
+import org.apache.pinot.segment.spi.index.IndexCreator;
+import org.apache.pinot.segment.spi.index.IndexHandler;
+import org.apache.pinot.segment.spi.index.IndexReader;
+import org.apache.pinot.segment.spi.index.IndexReaderFactory;
+import org.apache.pinot.segment.spi.index.IndexType;
+import org.apache.pinot.segment.spi.index.StandardIndexes;
+import org.apache.pinot.segment.spi.store.SegmentDirectory;
+import org.apache.pinot.spi.config.table.IndexConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.data.Schema;
+
+
+public class RangeIndexType implements IndexType<IndexConfig, IndexReader, IndexCreator> {
+
+  /**
+   * The default range index version used when not specified in the TableConfig.
+   *
+   * This value should be equal to the one used in {@link org.apache.pinot.spi.config.table.IndexingConfig}
+   */
+  public static final int DEFAULT_RANGE_INDEX_VERSION = 2;
```

Comment:
Not sure where this would be used.. and how we change the index version

Reply:
It will be used in the next PR. We have been talking about it [here](https://github.com/apache/pinot/pull/10193#discussion_r1146922455). 

TL;DR: The const is not used here, but I added the javadoc by mistake on this PR. We can either add here the const (no problematic), let the javadoc written without the attribute (which seems rare) or remove here the javadoc. The later may seem to be the best option, but if I do that, the javadoc will be removed from the full PR once I merge the changes. If I don't remember to add it later, we will lose the javadoc. Given that this process is error prone, I decided to just bring the const to this PR, which causes no problem at the cost of having a const we don't actually use here.

>  and how we change the index version
This is the range index version we use right now.